### PR TITLE
[Perf][Spec Decoding] Skip cat/topk/sort/gather in draft_forward for topk=1

### PR DESCRIPTION
**Pull request by Qiaolin-Yu**
When speculative_eagle_topk == 1 the draft tree is a chain, so the post-loop machinery in draft_forward — concat(score_list).flatten → torch.topk(.., num_draft-1) → torch.sort → torch.gather over ss_token_list — is structurally an identity:

  parent_list      = [-1, 0, 1, ..., num_draft-2]   (chain parents)
  top_scores_index = [ 0, 1, 2, ..., num_draft-2]   (sorted picks)
  draft_tokens     = cat(token_list, dim=1)

Pre-allocate parent_list / top_scores_index once at worker __init__ sized to cuda_graph_max_bs and slice [:bs] inside the captured graph. The bitonicSortKVInPlace + sbtopk::gatherTopK kernels disappear from the DRAFT_DECODE phase.

topk > 1 path is unchanged.

Measured: tightens DRAFT_DECODE GPU work by ~25 us/cycle. Combined with other topk=1 fast-paths the net was +2 tok/s on Kimi-K2.5-NVFP4 TP=4 80K ctx EAGLE3 topk=1 bs=1. gsm8k unchanged.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26780151731](https://github.com/sgl-project/sglang/actions/runs/26780151731)
Latest PR Test (Extra): :x: [Run #26780151701](https://github.com/sgl-project/sglang/actions/runs/26780151701)

### CODE REVIEW

File: python/sglang/srt/speculative/eagle_utils.py [hunk 1]
```diff
@@ -98,13 +99,21 @@ def organize_draft_results(
...
+    )
     draft_tokens = torch.gather(ss_token_list, index=top_scores_index, dim=1)
 
     if len(parents_list) > 1:
         parent_list = torch.cat(parents_list[:-1], dim=1)
     else:
         batch_size = parents_list[0].shape[0]
-        parent_list = torch.empty(batch_size, 0, device=parents_list[0].device)
+        parent_list = torch.empty(
+            batch_size, 0, dtype=torch.long, device=parents_list[0].device
```
Comment (kpham-sgl) [inline python/sglang/srt/speculative/eagle_utils.py:115]: curious why dtype=long here

File: python/sglang/srt/speculative/eagle_worker_v2.py [hunk 2]
```diff
@@ -213,6 +219,35 @@ def __init__(
 
         self.plan_stream, self.plan_stream_ctx = _get_plan_stream(self.device)
 
+    def _rebuild_topk1_chain_buffers(self) -> None:
+        # For topk=1 the draft tree degenerates to a chain, so parent_list and
+        # top_scores_index are runtime-invariant. Must be rebuilt after any
+        # change to speculative_num_steps / speculative_num_draft_tokens.
```
Comment (kpham-sgl) [inline python/sglang/srt/speculative/eagle_worker_v2.py:225]: This is because of adaptive spec dec right?

Comment (Qiaolin-Yu) [reply python/sglang/srt/speculative/eagle_worker_v2.py:225]: yes

File: python/sglang/srt/speculative/eagle_utils.py (hunk 1 above)
Comment (Qiaolin-Yu) [reply python/sglang/srt/speculative/eagle_utils.py:115]: https://github.com/sgl-project/sglang/blob/9a8ab2d22b4f85f840701e8ed717f663e1c4662b/sgl-kernel/csrc/speculative/eagle_utils.cu#L242

It's actually used as long in kernel.

File: test/registered/unit/spec/test_eagle_worker_v2_topk1_fastpath.py
Comment (kpham-sgl) [inline test/registered/unit/spec/test_eagle_worker_v2_topk1_fastpath.py:1]: Btw I think there are some new Eagle Harnesses for unit test. Not sure if any of them applies here

**Comment by Qiaolin-Yu** [comment]
/rerun-test test/registered/8-gpu-models/test_step3p5_flash_chain_mtp.py

**Comment by Qiaolin-Yu** [comment]
/rerun-test test/registered/models_e2e/test_mimo_v2_flash.py test/registered/models_e2e/test_step3p5_flash_chain_mtp.py

**Review by kpham-sgl — APPROVED** [review]
